web/src/invite.ts

```diff
@@ -44,6 +44,18 @@ let stream_pill_widget: stream_pill.StreamPillWidget;
 let user_group_pill_widget: user_group_pill.UserGroupPillWidget;
 let guest_invite_stream_ids: number[] = [];
 
+type CommonInvitationData = {
+    csrfmiddlewaretoken: string;
+    invite_as: number;
+    notify_referrer_on_join: boolean;
+    stream_ids: string;
+    group_ids: string;
+    invite_expires_in_minutes: string;
+    invitee_emails: string;
+    include_realm_default_subscriptions: string;
+    welcome_message_custom_text?: string;
+};
```

Comment:
The refactor to introduce this type would be nicely done as a prep commit, then the main commit just adds a thing or two to the type.